src/index.js

```diff
@@ -35,14 +36,9 @@ export default function supernova(env) {
       const theme = useTheme();
 
       const [pageInfo, setPageInfo] = useState({ top: 0, height: 100 });
-      const [tableData, setTableData] = useState();
       const [muiParameters] = useState(muiSetup(__OPIONAL_THEME_DEPS__));
 
-      useEffect(() => {
-        manageData(model, layout, pageInfo).then((d) => {
-          setTableData(d);
-        });
-      }, [layout, pageInfo]);
+      const [tableData] = usePromise(() => manageData(model, layout, pageInfo), [layout, pageInfo]);
```

Comment:
Don't remember how `usePromise` works. `tableData` will update when `layout` or `pageInfo` changes?

Reply:
hmnm, didn't acutally test, seems not, I'll check

Reply:
yea it works as expected afaik

Reply:
I think it workss the same as before but more concisely

Reply:
https://qlik.dev/apis/javascript/nebulajs-stardust#%23%2Fentries%2FusePromise the example here is not the best I would say